### PR TITLE
Pd js2 w1

### PR DESCRIPTION
**Pull request by esma-g**
## What does this change?

Module: JS2
Week(s): 1

## Checklist

- [x] I have read the [contributing guidelines](https://syllabus.codeyourfuture.io/contributing/overview)
- [x] I have checked my spelling and grammar with an [automated tool](https://www.grammarly.com/grammar-check)
- [x] I have previewed my changes to check the [markdown renders](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) as I intend
- [x] I have run my code to check it works
- [x] My changes follow our [Style Guide](https://syllabus.codeyourfuture.io/guides/code-style-guide)

## Description

Added PD lesson and coursework. Editing week 2-4 by mistake in this PR, removed them later on.

## Who needs to know about this?

@kfklein15 @CodeYourFuture/syllabus-team

### CODE REVIEW

File: docs/js-core-2/week-1/lesson.md [hunk 1]
```diff
@@ -1258,6 +1259,165 @@ console.log(
...
+- Tech lead 
+- Team lead / Manager
+
+**Answer key**
+
+Brief descriptions of the roles:
+
+- A front end web developer: developer that mainly works on the look and feel of the website 
+- A back end web developer: developer that mainly works on the server-side code (database, API, etc) 
+- A UX/UI designer or UX researcher: Provides the design of the website
```
Comment (kfklein15) [inline docs/js-core-2/week-1/lesson.md]: design and user experience

File: docs/js-core-2/week-1/lesson.md [hunk 2]
```diff
@@ -1258,6 +1259,165 @@ console.log(
...
+- A Product manager: Sets work priorities, and product direction 
+- A QA tester: quality assurance team member. Their responsibility is to test the work that developers produce, making sure it works well, with no defects or missing functionality.
+- Tech lead: Hands-on developer responsible for tech excellence, code best practices (no people management responsibilities)
+- Team lead / Manager: Hands-on developer responsible for the wellbeing of the team, and team morale (sometimes has people management responsibilities)
```
Comment (kfklein15) [inline docs/js-core-2/week-1/lesson.md]: Hmmm not sure this one makes sense. I would stop at Tech Lead

File: docs/js-core-2/week-1/lesson.md [hunk 3]
```diff
@@ -1258,6 +1259,165 @@ console.log(
...
+
+Discuss as a class the responsibilities of the following tech roles:
+
+- A front end web developer
+- A back end web developer 
+- A UX/UI designer or UX researcher 
+- A Product manager
+- A QA tester 
+- Tech lead 
+- Team lead / Manager
```
Comment (kfklein15) [inline docs/js-core-2/week-1/lesson.md]: This one doesn't make sense to me

**Review by kfklein15 — CHANGES_REQUESTED** [review]
Asked for minor changes

File: docs/js-core-2/week-1/lesson.md [hunk 4]
```diff
@@ -1258,6 +1259,165 @@ console.log(
...
+- Prepare a brief for your project
+
+**Session objective**: In this session, we will talk about tech teams and different tech roles. We will briefly discuss different aspects of teamwork; such as setting up goals, software development processes, time management, communication, and conflict management.
+
+:::note
+Exercise (10 minutes)
+
+**Exercise objective**: To brainstorm about teamwork at tech industry
+
+Think about the teams you will work with in future when you move in tech. Discuss in groups:
```
Comment (illicitonion) [inline docs/js-core-2/week-1/lesson.md]:
```suggestion
Think about the teams you will work with in the future when you move into tech. Discuss in groups:
```

File: docs/js-core-2/week-1/lesson.md [hunk 5]
```diff
@@ -1258,6 +1259,165 @@ console.log(
...
+- Why is it important for everybody on your team to understand the role each other play?
+- What happens when you are not clear about these two points?
+
+:::
+
+:::note
+Exercise (5 minutes)
+
+**Exercise objective:** To get familiar with different processes in the software development cycle
+Match the process to the stage in the software development cycle (SDC) as a class. Don’t go too much into detail, you can do some research later if you would like to learn more. These processes vary from company to company, and determine what steps need to happen before code is shipped to production. They aim to reduce the risk of mistakes going to production. For example during the deployment process, code is reviewed by at least one person, deployed to a testing environment (server), successfully tested by a QA, and the release is scheduled to a time approved by the end customer.
```
Comment (illicitonion) [inline docs/js-core-2/week-1/lesson.md]: I'm more familiar with the term SDLC - Software Development Lifecycle; is SDC commonly used too?

File: docs/js-core-2/week-1/lesson.md [hunk 6]
```diff
@@ -1258,6 +1259,165 @@ console.log(
...
+For the second question the correct answer at CYF is “GitHub Projects”.
+Some examples for tools: Jira, GitHub (repo and project boards), IDE (integrated development environment), Slack, Teams, Google calendar, Calendly, Microsoft Office, Outlook, shared documents, Miro, Figma, Jamboard
+
+:::
+
+:::note
+Exercise (10 minutes)
+
+**Exercise objective:** To brainstorm about the causes of conflicts and methods to manage them
+Remember, there will be a more detailed PD session on conflict resolution in the upcoming weeks. Work in groups for about 5 minutes, then discuss as a class for 5 minutes:
```
Comment (illicitonion) [inline docs/js-core-2/week-1/lesson.md]:
```suggestion
**Exercise objective:** To brainstorm about the causes of conflicts and methods to manage them

Remember, there will be a more detailed PD session on conflict resolution in the upcoming weeks. Work in groups for about 5 minutes, then discuss as a class for 5 minutes:
```

If there's not an empty line between two lines in markdown, the two lines get folded into one.

File: docs/js-core-2/week-3/lesson.md [hunk 7]
```diff
@@ -20,7 +20,7 @@ import Feedback from "@theme/Feedback";
     - [Exercise (4)](#exercise-4)
   - [Manipulate DOM elements](#manipulate-dom-elements)
     - [Exercise (5)](#exercise-5)
-
+  - [PD - Delivery](#pd-delivery)
```
Comment (illicitonion) [inline docs/js-core-2/week-3/lesson.md]: Accidentally added in this PR?

File: docs/js-core-2/week-4/lesson.md [hunk 8]
```diff
@@ -18,6 +18,7 @@ import Feedback from "@theme/Feedback";
   - [Client/Server architecture](#clientserver-architecture)
   - [HTTP Requests](#http-requests)
     - [Exercise (2)](#exercise-2)
+  - [PD - Presentations](#pd-presentations)
```
Comment (illicitonion) [inline docs/js-core-2/week-4/lesson.md]: Accidentally added in this PR?

**Review by illicitonion — COMMENTED** [review]
Looks great, just a few minor things - thanks!

File: docs/js-core-2/week-1/lesson.md (hunk 5 above)
Comment (esma-g) [reply docs/js-core-2/week-1/lesson.md]: The developers in the PD syllabus team suggested SDC. Would you like me to change @illicitonion ?

File: docs/js-core-2/week-4/lesson.md (hunk 8 above)
Comment (esma-g) [reply docs/js-core-2/week-4/lesson.md]: Yes, sorry. fixed

File: docs/js-core-2/week-1/lesson.md (hunk 1 above)
Comment (esma-g) [reply docs/js-core-2/week-1/lesson.md]: fixed

File: docs/js-core-2/week-1/lesson.md (hunk 2 above)
Comment (esma-g) [reply docs/js-core-2/week-1/lesson.md]: fixed

File: docs/js-core-2/week-1/lesson.md (hunk 3 above)
Comment (esma-g) [reply docs/js-core-2/week-1/lesson.md]: fixed

File: docs/js-core-2/week-3/lesson.md (hunk 7 above)
Comment (esma-g) [reply docs/js-core-2/week-3/lesson.md]: Yes, sorry. fixed

File: docs/js-core-2/week-1/lesson.md (hunk 5 above)
Comment (illicitonion) [reply docs/js-core-2/week-1/lesson.md]: Yeah, I'd go with SDLC if that's ok, sorry!

**Review by illicitonion — DISMISSED** [review]

**Comment by illicitonion** [comment]
I just force-pushed resolving all the conflicts from the other PRs, as well as applying the SDC -> SDLC update - hope you don't mind!

If you wanted to make more edits, you should pull down the changes before you edit more :)

**Review by illicitonion — APPROVED** [review]

**Comment by esma-g** [comment]
@kfklein15 Could I get a review for this one, please?

**Review by kfklein15 — APPROVED** [review]
lgtm

**Pull request merged**